src/modules/flow/http-client/http-client.c

```diff
@@ -714,6 +714,276 @@ float_post_process(struct sol_flow_node *node, void *data, uint16_t port, uint16
+}
+
+static int
+rgb_post_process(struct sol_flow_node *node, void *data, uint16_t port,
+    uint16_t conn_id, const struct sol_flow_packet *packet)
+{
+    int r;
+    struct sol_rgb rgb;
+
+#define INT32_STR_LEN (12)
+
+    SOL_BUFFER_DECLARE_STATIC(red, INT32_STR_LEN);
+    SOL_BUFFER_DECLARE_STATIC(green, INT32_STR_LEN);
+    SOL_BUFFER_DECLARE_STATIC(blue, INT32_STR_LEN);
+    SOL_BUFFER_DECLARE_STATIC(red_max, INT32_STR_LEN);
+    SOL_BUFFER_DECLARE_STATIC(green_max, INT32_STR_LEN);
+    SOL_BUFFER_DECLARE_STATIC(blue_max, INT32_STR_LEN);
+
+    r = sol_flow_packet_get_rgb(packet, &rgb);
+    SOL_INT_CHECK(r, < 0, r);
+
+    r = sol_buffer_append_printf(&red, "%" PRIu32, rgb.red);
+    SOL_INT_CHECK(r, < 0, r);
+    r = sol_buffer_append_printf(&green, "%" PRIu32, rgb.green);
+    SOL_INT_CHECK(r, < 0, r);
+    r = sol_buffer_append_printf(&blue, "%" PRIu32, rgb.blue);
+    SOL_INT_CHECK(r, < 0, r);
+    r = sol_buffer_append_printf(&red_max, "%" PRIu32, rgb.red_max);
+    SOL_INT_CHECK(r, < 0, r);
+    r = sol_buffer_append_printf(&green_max, "%" PRIu32, rgb.green_max);
+    SOL_INT_CHECK(r, < 0, r);
+    r = sol_buffer_append_printf(&blue_max, "%" PRIu32, rgb.blue_max);
+    SOL_INT_CHECK(r, < 0, r);
+
+    return common_post_process(node, data, "red", red.data,
+        "green", green.data, "blue", blue.data, "red_max", red_max.data,
+        "green_max", green_max.data, "blue_max", blue_max.data, NULL);
+
+#undef INT32_STR_LEN
+}
+
+/*
+ * --------------------------------- direction vector node  -----------------------------
+ */
+static int
+direction_vector_process_token(struct sol_flow_node *node,
+    struct sol_json_token *key, struct sol_json_token *value)
+{
+    struct sol_direction_vector dir_vector = { 0 };
+    enum sol_json_loop_reason reason;
+    struct sol_json_scanner sub_scanner;
+    struct sol_json_token sub_key, sub_value, token;
+
+    sol_json_scanner_init(&sub_scanner, value->start,
+        value->end - value->start);
+    SOL_JSON_SCANNER_OBJECT_LOOP (&sub_scanner, &token, &sub_key,
+        &sub_value, reason) {
+        if (SOL_JSON_TOKEN_STR_LITERAL_EQ(&sub_key, "x")) {
+            if (sol_json_token_get_double(&sub_value, &dir_vector.x) < 0)
+                return -EINVAL;
+        } else if (SOL_JSON_TOKEN_STR_LITERAL_EQ(&sub_key, "y")) {
+            if (sol_json_token_get_double(&sub_value, &dir_vector.y) < 0)
+                return -EINVAL;
+        } else if (SOL_JSON_TOKEN_STR_LITERAL_EQ(&sub_key, "z")) {
+            if (sol_json_token_get_double(&sub_value, &dir_vector.z) < 0)
+                return -EINVAL;
+        } else if (SOL_JSON_TOKEN_STR_LITERAL_EQ(&sub_key, "min")) {
+            if (sol_json_token_get_double(&sub_value, &dir_vector.min) < 0)
+                return -EINVAL;
+        } else if (SOL_JSON_TOKEN_STR_LITERAL_EQ(&sub_key, "max")) {
+            if (sol_json_token_get_double(&sub_value, &dir_vector.max) < 0)
+                return -EINVAL;
+        }
+    }
```

Comment:
same comment I did for rgb, we should have better `min` and `max` defaults, check them before dispatching the packet.